server/internal/mcp/impl.go

```diff
@@ -404,6 +455,19 @@ func (s *Service) ServePublic(w http.ResponseWriter, r *http.Request) error {
 				Token:        externalSecret.Token,
 			})
 		}
+	case toolset.McpIsPublic && hasExternalMCPOAuth:
```

Comment:
I already thought all this oauth stuff needed to be factored out of this method even before this addition. Let's figure out a way to encapsulate the oauth logic elsewhere. At the very least make a ticket with a link to this method saying to refactor